restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -576,13 +600,18 @@ private void validateOpAndValue(String op, JsonNode value, String fieldName) {
   }
 
   public void deleteAtPath(
-      DocumentDB db, String keyspace, String collection, String id, List<PathSegment> path) {
+      DocumentDB db, String keyspace, String collection, String id, List<PathSegment> path)
+      throws UnauthorizedException {
     List<String> convertedPath = new ArrayList<>(path.size());
     for (PathSegment pathSegment : path) {
       String pathStr = pathSegment.getPath();
       convertedPath.add(convertArrayPath(pathStr));
     }
     Long now = ChronoUnit.MICROS.between(Instant.EPOCH, Instant.now());
+
+    db.getAuthorizationService()
+        .authorizeDataWrite(db.getAuthToken(), keyspace, collection, Scope.DELETE);
+
     db.delete(keyspace, collection, id, convertedPath, now);
```

Comment:
Would it make sense to move the `authorizeDataWrite` call inside `db.delete` to have to apply to all `delete` cases (essentially reducing the number of places where `authorizeDataWrite` is invoked)?

Reply:
I wonder if it makes sense to do this in general? That is, to push all the authz into the db layer. It has access to the authz service and auth token already and it's aware of the type of modification being made. It might be cleaner and more potentially less error prone.

Reply:
Fixed.